crates/emitter/src/ast_emitter.rs

```diff
@@ -444,25 +444,41 @@ impl<'alloc, 'opt> AstEmitter<'alloc, 'opt> {
                     expression,
                     ..
                 },
-            )) => {
-                self.emit_expression(expression)?;
-
-                match property_name {
-                    PropertyName::StaticPropertyName(StaticPropertyName { value, .. }) => {
-                        let name_index = self.emit.get_atom_index(*value);
-                        self.emit.init_prop(name_index);
-                    }
-                    PropertyName::ComputedPropertyName(ComputedPropertyName { .. }) => {
-                        return Err(EmitError::NotImplemented("TODO: computed property"))
+            )) => match property_name {
+                PropertyName::StaticPropertyName(StaticPropertyName { value, .. }) => {
+                    match self.to_property_index(*value) {
+                        Some(value) => {
+                            self.emit.double_(value as f64);
+                            self.emit_expression(expression)?;
+                            self.emit.init_elem();
+                        }
+                        None => {
+                            self.emit_expression(expression)?;
+                            let name_index = self.emit.get_atom_index(*value);
+                            self.emit.init_prop(name_index);
+                        }
                     }
                 }
-            }
+                PropertyName::StaticNumericPropertyName(NumericLiteral { value, .. }) => {
+                    self.emit.double_(*value);
+                    self.emit_expression(expression)?;
+                    self.emit.init_elem();
```

Comment:
nit: Factor these out in a new function?

Reply:
This will be handled by adding `ObjectEmitter` helper struct.